src/pages/movies/[movieId].js

```diff
@@ -0,0 +1,175 @@
+import styles from "@/styles/movie.module.scss";
+import { getMovie, getShowtimesForMovie } from "@/utils/dbFunctions";
+import dbConnect from "@/utils/dbConnect";
+import Link from "next/link";
+
+export async function getServerSideProps(context) {
+  await dbConnect();
+
+  const { movieId } = context.query;
+  const movie = (await getMovie(movieId)) ?? null;
+
+  if (!movie) {
+    return {
+      notFound: true
+    };
+  }
+
+  const showtimes = await getShowtimesForMovie(movieId);
+
+  return {
+    props: {
+      showtimes: JSON.parse(JSON.stringify(showtimes)).sort(
```

Comment:
maybe it would be better to display showtimes from todays date and not the ones that have passed (right now you get all the showtimes and someone can "book" a showing that was two weeks ago). 
And also, it would be maybe nice to limit the output to 10-15 showtimes, so it doesnt take up so much space! :)

Reply:
Good suggestion, I've updated the db query to only show coming showtimes.
Your second suggestion is also good, I would like to implement pagination with more time. Alas, not today!